Sources/BraintreeSEPADirectDebit/CreateMandateResult.swift

```diff
@@ -39,7 +39,9 @@ struct CreateMandateResult: Decodable {
         let bodyContainer = try messageContainer.nestedContainer(keyedBy: CodingKeys.self, forKey: .body)
         let sepaDebitContainer = try bodyContainer.nestedContainer(keyedBy: CodingKeys.self, forKey: .sepaDebitAccount)
         
-        approvalURL = try sepaDebitContainer.decode(String.self, forKey: .approvalURL)
+        // Defaulting the approval URL to the string "null" if the API returns nil for this field because the
+        // mandate has already been approved.
+        approvalURL = try sepaDebitContainer.decodeIfPresent(String.self, forKey: .approvalURL) ?? "null"
```

Comment:
Ah I missed this the first time - but we should probably create the constant for `"null"` in this class. Then we should reference that constant when we check it's value later. That way if for any reason this `"null"` string is ever changed to something else, the things that check it would be automatically changed to check the correct value as well.

Reply:
Could we make `approvalURL` optional here instead of having `"null"`? Or maybe even an empty `""` would work well.

Reply:
I went back and fourth on this, on Android, the literal string returned  from the API is "null" which is what we do the check for on Android. I think we could make it optional and then do a nil check, but then it would be hard to handle if something went wrong and the result of the URL was nil vs now what we do is get "null" from the API, which Swift converts to nil, then we convert back to "null" in the decoding step. It could potentially leave us in a situation when the URL is unexpectedly actually nil but we handle it as if the mandate was already approved, which would be incorrect.

Reply:
Nice yeah I agree with Kayla here then. If we put it in a constant and maybe add a comment with this information right above it to keep the historical context. It'll help us cut down on WTFs/min lol.